### ISSUE TITLE
`object-shorthand` and "unexpected token }" (A parser bug?)

### ISSUE THREAD

**Issue by slacktracer** [eslint/espree#183]
I opened this issue on eslint: https://github.com/eslint/eslint/issues/3333

I was told to report here. I don't know the inner workings around here. I decided to do as told. Correct me if I am wrong.

---

Having this piece of code:

    // test.js
    'use strict';
    function f(value) {
        return {
            value
        };
    }
    let o = f('some value');
    console.log(o);

And this .eslintrc:

    {
        "rules": {
            "object-shorthand": [
                2,
                "always"
            ]
        }
    }

I get the following:

      5:6  error  Unexpected token }

[I asked about it on StackOverflow.](http://stackoverflow.com/questions/31898616/identifying-gulp-eslint-error-reason)

Is it a bug? What can I do to verify it and help?

Also, if I change `f` to:

    function f(value) {
        var property = true;
        return {
            value,
            property
        };
    }

I get:

      5:15  error  Unexpected token ,

And if I change it to:

    function f(value) {
        let property = true;
        return {
            value,
            property
        };
    }

I get:

      3:6  error  Use of future reserved word in strict mode

It's starting to weird me out. =P

(I'm using eslint v1.1.0.)

**Comment by xjamundx** [comment]
I can't reproduce this, you probably just need to specify your the es6 environment in your eslint config file.


```
> require('espree').parse("'use strict';function f(value) {    return {        value    };}let o = f('some value');console.log(o);", {ecmaFeatures: {  blockBindings: true,         objectLiteralShorthandProperties: true, }})
```

```js
{ type: 'Program',
  body: 
   [ { type: 'ExpressionStatement', expression: [Object] },
     { type: 'FunctionDeclaration',
       id: [Object],
       params: [Object],
       body: [Object],
       generator: false,
       expression: false },
     { type: 'VariableDeclaration',
       declarations: [Object],
       kind: 'let' },
     { type: 'ExpressionStatement', expression: [Object] } ],
  sourceType: 'script' }
```
```
> require('espree').parse("function f(value) {    let property = true;    return {        value,        property    };}", {ecmaFeatures: {  blockBindings: true,         objectLiteralShorthandProperties: true, }})
```

```js
{ type: 'Program',
  body: 
   [ { type: 'FunctionDeclaration',
       id: [Object],
       params: [Object],
       body: [Object],
       generator: false,
       expression: false } ],
  sourceType: 'script' }
```

**Comment by slacktracer** [comment]
Thank you, @xjamundx 

I did not know. I thought setting the rule was enough.
